### ISSUE TITLE
ReadCancellationException does not play well with `onErrorContinue`

### ISSUE THREAD

**Issue by chringwer** [spring-projects/spring-framework#24125]
The way that `ReadCancellationException` is used in `BodyExtractors` is apparently meant to be an implementation detail: 
https://github.com/spring-projects/spring-framework/blob/1bff7ce1418807d1f75daa9025fabfe4d4203bb9/spring-webflux/src/main/java/org/springframework/web/reactive/function/BodyExtractors.java#L269-L273
However, if somewhere downstream in the pipeline an `onErrorContinue` operator is appended, the exception does surface which caused me some confusion. Could this be implemented differently without throwing an exception?

**Comment by rstoyanchev** [comment]
Good point that a downstream `onErrorContinue` can quietly overtake `onErrorResume` handling and preclude buffer handling. Out of curiosity how did you even come across the issue?

We'll explore some alternative. We need to review similar handling elsewhere (e.g. `DefaultWebClient`).

**Comment by chringwer** [comment]
I ran into it when using WebClient as a reactive source in spring cloud stream. Under the hood it was `FluxMessageChannel` from spring integration that registered the `onErrorContinue` operator here:
https://github.com/spring-projects/spring-integration/blob/v5.2.1.RELEASE/spring-integration-core/src/main/java/org/springframework/integration/channel/FluxMessageChannel.java#L85

But apparently, that has already been fixed by a commit a few minutes ago. :-)

However, thanks for looking into this!
